pyvista/plotting/render_passes.py

```diff
@@ -0,0 +1,255 @@
+    def _seq_pass(self):
+        """Initialize (when necessary the pass collection and return it.
+
+        This lets us lazily generate the pass collection only when we need it
+        rather than at initialization of the class.
+
+        """
+        if self.__seq_pass is None:
+            self._init_passes()
+        return self.__seq_pass
+
+    @property
+    def _camera_pass(self):
+        """Initialize (when necessary the pass collection and return it.
+
+        This lets us lazily generate the pass collection only when we need it
+        rather than at initialization of the class.
+
+        """
+        if self.__camera_pass is None:
+            self._init_passes()
+        return self.__camera_pass
+
+    def _init_passes(self):
+        """Initialize the renderer's standard passes."""
+        # simulate the standard VTK rendering passes and put them in a sequence
+        self.__pass_collection = _vtk.vtkRenderPassCollection()
+        self.__pass_collection.AddItem(_vtk.vtkRenderStepsPass())
+
+        self.__seq_pass = _vtk.vtkSequencePass()
+        self.__seq_pass.SetPasses(self._pass_collection)
+
+        # Make the sequence the delegate of a camera pass.
+        self.__camera_pass = _vtk.vtkCameraPass()
+        self.__camera_pass.SetDelegatePass(self._seq_pass)
+
+    @property
+    def _renderer(self):
+        """Return the renderer."""
+        if self._renderer_ref is not None:
+            return self._renderer_ref()
+
+    def deep_clean(self):
+        """Delete all render passes."""
+        if self._renderer is not None:
+            self._renderer.SetPass(None)
+        self._renderer_ref = None
+        if self.__seq_pass is not None:
+            self.__seq_pass.SetPasses(None)
+        self.__seq_pass = None
+        self.__pass_collection = None
+        self.__camera_pass = None
+        self._passes = {}
+        self._shadow_map_pass = None
+        self._edl_pass = None
+        self._dof_pass = None
+        self._ssaa_pass = None
+        self._blur_passes = []
+
+    def enable_edl_pass(self):
+        """Enable the EDL pass."""
+        if self._edl_pass is not None:
+            return
+        self._edl_pass = _vtk.vtkEDLShading()
+        self._add_pass(self._edl_pass)
+        return self._edl_pass
+
+    def disable_edl_pass(self):
+        """Disable the EDL pass."""
+        if self._edl_pass is None:
+            return
+        self._remove_pass(self._edl_pass)
+        self._edl_pass = None
+
+    def add_blur_pass(self):
+        """Add a vtkGaussianBlurPass pass.
+
+        This is an vtkImageProcessingPass and delegates to the last pass.
+
+        """
+        blur_pass = _vtk.vtkGaussianBlurPass()
+        self._add_pass(blur_pass)
+        self._blur_passes.append(blur_pass)
+        return blur_pass
+
+    def remove_blur_pass(self):
+        """Add a vtkGaussianBlurPass pass."""
+        if self._blur_passes:
+            self._remove_pass(self._blur_passes.pop(0))
```

Comment:
When we add blur passes we append. Does it make sense to `pop(0)` when we remove? Shouldn't we pop off the most recent one? I have no idea how this works, sorry...

Reply:
I agree that it should be LIFO

Reply:
Order of the blur passes doesn't matter. Added note in 69243a9ff453f7d292ce9329026b0ef8c43fa355.